server/supernova/sc/sc_synthdef.cpp

```diff
@@ -381,8 +381,12 @@ void sc_synthdef::prepare(void)
     memory_requirement_ += (graph.size() + calc_unit_indices.size() + 1) * sizeof(Unit*); // reserves space for units (one more to allor prefetching)
 
     // memory that is required to fill the sc_synth data structure
-    const size_t ctor_alloc_size = parameter_count() * (sizeof(float) + sizeof(int) + sizeof(float*))
-                                 + constants.size() * sizeof(Wire);
+    // see SC_Graph.h for corresponding data structure
+    // mControls, mControlRates, mAudioBusOffset, mMapControls
+    const size_t param_alloc_size = parameter_count() * (sizeof(float) + sizeof(int) + sizeof(int32) + sizeof(float*));
```

Comment:
This is a really good find!(!) I worry that we may forget about this in the future, too - is there somewhere we can document this sort of thing?

Reply:
@vivid-synth are the notes around the related code not enough? Where else could I document it? Currently AFAIK (and it seems like I'm involved in most of these conversations), there is no real documentation of these internals. I would be happy to write more verbose in-source comments if you think that's a good idea. I have done some work on getting Doxygen hooked up with our CI to automate online docs,
 and I was envisioning using that at some point in the future when it actually becomes worthwhile to direct people there. (http://www.brianlheim.com/supercollider/)

Reply:
Oh I missed that you've got docs on the ```.h``` and ```.cpp``` files too, nice 👍